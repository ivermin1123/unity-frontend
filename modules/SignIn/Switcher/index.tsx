import React, { useEffect, useState } from 'react';
import { useTheme as useNextTheme } from 'next-themes';
import { Row, Switch, useTheme } from '@nextui-org/react';
import { AnimatePresence, motion } from 'framer-motion';

import LightIcon from '@/public/icons/light.svg';
import DarkIcon from '@/public/icons/dark.svg';

const Switcher = () => {
  const { setTheme } = useNextTheme();
  const { isDark, theme } = useTheme();

  const [mounted, setMounted] = useState(false);

  // useEffect only runs on the client, so now we can safely show the UI
  useEffect(() => {
    setMounted(true);
  }, []);

  if (!mounted) {
    return null;
  }

  return (
    <Row
      css={{
        alignSelf: 'flex-end',
        alignItems: 'center',
        justifyContent: 'flex-start',
        gap: theme?.space[8],
        marginLeft: theme?.space[18],
      }}
    >
      <motion.div
        initial={{
          display: !isDark ? 'flex' : 'none',
        }}
        animate={{
          opacity: Number(!isDark),
          transitionEnd: { display: !isDark ? 'flex' : 'none' },
        }}
        exit={{ opacity: 0 }}
      >
        <LightIcon />
      </motion.div>

      <motion.div
        initial={{
          display: isDark ? 'flex' : 'none',
        }}
        animate={{
          opacity: Number(isDark),
          transitionEnd: { display: isDark ? 'flex' : 'none' },
        }}
        exit={{ opacity: 0 }}
      >
        <DarkIcon />
      </motion.div>

      <Switch
        checked={isDark}
        onChange={e => setTheme(e.target.checked ? 'dark' : 'light')}
      />
    </Row>
  );
};

export default Switcher;
