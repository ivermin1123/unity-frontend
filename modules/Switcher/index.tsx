import { CSS, Row, Switch, useTheme } from '@nextui-org/react';
import { useTheme as useNextTheme } from 'next-themes';
import { useEffect, useState } from 'react';
import { BiMoon, BiSun } from 'react-icons/bi';

type SwitcherProps = {
  style?: CSS;
};

const Switcher = ({ style }: SwitcherProps) => {
  const { setTheme } = useNextTheme();
  const { isDark, theme } = useTheme();

  const [mounted, setMounted] = useState(false);

  // useEffect only runs on the client, so now we can safely show the UI
  useEffect(() => {
    setMounted(true);
  }, []);

  if (!mounted) {
    return null;
  }

  return (
    <Row
      css={{
        alignSelf: 'flex-end',
        alignItems: 'center',
        gap: theme?.space[8],
        marginTop: 'auto',
        ...style,
      }}
    >
      <Switch
        data-testid="switcher"
        shadow
        checked={isDark}
        onChange={e => setTheme(e.target.checked ? 'dark' : 'light')}
        iconOff={<BiSun fill="black" color="black" />}
        iconOn={<BiMoon />}
      />
    </Row>
  );
};

export default Switcher;
