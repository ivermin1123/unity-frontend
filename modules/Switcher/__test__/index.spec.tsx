import { act, render } from '@shared/utils/test/test-ultis';
import Switcher from '../index';

describe('Switcher', () => {
  it('should render correctly', async () => {
    await act(async () => {
      await new Promise(resolve => {
        setTimeout(resolve, 50);
      });
    });
    const { container } = render(<Switcher />);

    expect(container).toMatchSnapshot();
  });
});
