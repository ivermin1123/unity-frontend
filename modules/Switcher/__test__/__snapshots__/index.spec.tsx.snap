// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher should render correctly 1`] = `
<div>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light-theme','dark-theme');var e=localStorage.getItem('theme');if(e){var x={"light":"light-theme","dark":"dark-theme"};c.add(x[e]|| '')}else{c.add('light-theme');}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(t){}}();
  </script>
  <div
    data-overlay-container="true"
  >
    <div
      class="nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-igaDGwc-css"
    >
      <label
        class="nextui-c-STltJ nextui-c-STltJ-bwQcWP-color-default nextui-c-STltJ-JJjKI-size-md nextui-c-STltJ-bgIFDi-borderWeight-normal"
        data-state="unchecked"
        data-testid="switcher"
      >
        <input
          class="nextui-c-jjKPqy nextui-c-iFHsrU nextui-c-jjKPqy-gwHlSF-animated-true nextui-switch-input"
          data-state="unchecked"
          tabindex="-1"
          type="checkbox"
        />
        <div
          aria-checked="false"
          aria-disabled="false"
          class="nextui-c-itiWTf nextui-c-huiNHE nextui-c-itiWTf-iXJPXp-animated-true nextui-switch nextui-switch--unchecked"
          data-state="unchecked"
          role="switch"
          tabindex="0"
        >
          <span
            class="nextui-c-eJPKci nextui-switch-circle"
          >
            <svg
              color="black"
              fill="black"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: black;"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.993 12c0 2.761 2.246 5.007 5.007 5.007s5.007-2.246 5.007-5.007S14.761 6.993 12 6.993 6.993 9.239 6.993 12zM12 8.993c1.658 0 3.007 1.349 3.007 3.007S13.658 15.007 12 15.007 8.993 13.658 8.993 12 10.342 8.993 12 8.993zM10.998 19h2v3h-2zm0-17h2v3h-2zm-9 9h3v2h-3zm17 0h3v2h-3zM4.219 18.363l2.12-2.122 1.415 1.414-2.12 2.122zM16.24 6.344l2.122-2.122 1.414 1.414-2.122 2.122zM6.342 7.759 4.22 5.637l1.415-1.414 2.12 2.122zm13.434 10.605-1.414 1.414-2.122-2.122 1.414-1.414z"
              />
            </svg>
          </span>
        </div>
      </label>
    </div>
  </div>
</div>
`;
